05_deploy_bootstrap_vm.sh

```diff
@@ -78,59 +78,39 @@ sudo virt-install --connect qemu:///system \
+# NOTE: This is equivalent to the external API DNS record pointing the API to the API VIP
+IP=$(domain_net_ip ${CLUSTER_NAME}-bootstrap baremetal)
+API_IP=$(dig +noall +answer "${CLUSTER_NAME}-api.${BASE_DOMAIN}" @$(network_ip baremetal) | awk '{print $NF}')
+echo "address=/${CLUSTER_NAME}-api.${BASE_DOMAIN}/${API_IP}" | sudo tee /etc/NetworkManager/dnsmasq.d/openshift.conf
 sudo systemctl reload NetworkManager
 
 # Wait for ssh to start
 while ! ssh -o "StrictHostKeyChecking=no" core@$IP id ; do sleep 5 ; done
 
-# Using 172.22.0.1 on the provisioning network for PXE and the ironic API
-echo -e "DEVICE=eth1\nONBOOT=yes\nTYPE=Ethernet\nBOOTPROTO=static\nIPADDR=172.22.0.1\nNETMASK=255.255.255.0" | ssh -o "StrictHostKeyChecking=no" core@$IP sudo dd of=/etc/sysconfig/network-scripts/ifcfg-eth1
-ssh -o "StrictHostKeyChecking=no" core@$IP sudo ifup eth1
-
-# Internal dnsmasq should reserve IP addresses for each master
-cp -f ironic/dnsmasq.conf /tmp
-for i in 0 1 2; do
-  NODE_MAC=$(cat "${WORKING_DIR}/ironic_nodes.json" | jq -r ".nodes[${i}].ports[0].address")
-  NODE_IP="172.22.0.2${i}"
-  HOSTNAME="${CLUSTER_NAME}-etcd-${i}.${BASE_DOMAIN}"
-  # Make sure internal dnsmasq would assign an expected IP
-  echo "dhcp-host=${NODE_MAC},${HOSTNAME},${NODE_IP}" >> /tmp/dnsmasq.conf
-  # Reconfigure "external" dnsmasq
-  echo "${NODE_IP} ${HOSTNAME} ${CLUSTER_NAME}-api.${BASE_DOMAIN}" | sudo tee -a /etc/hosts.openshift
-  echo "srv-host=_etcd-server-ssl._tcp.${CLUSTER_NAME}.${BASE_DOMAIN},${HOSTNAME},2380,0,0" | sudo tee -a /etc/NetworkManager/dnsmasq.d/openshift.conf
-done
-sudo systemctl reload NetworkManager
-cat /tmp/dnsmasq.conf | ssh -o "StrictHostKeyChecking=no" core@$IP sudo dd of=dnsmasq.conf
+# ironic dnsmasq
+cat ironic/dnsmasq.conf | ssh -o "StrictHostKeyChecking=no" core@$IP sudo dd of=dnsmasq.conf
 
 # Build and start the ironic container
 cat "${RHCOS_IMAGE_FILENAME_OPENSTACK}" | ssh -o "StrictHostKeyChecking=no" "core@$IP" sudo dd of="${RHCOS_IMAGE_FILENAME_OPENSTACK}"
-
 IRONIC_IMAGE=${IRONIC_IMAGE:-"quay.io/metalkube/metalkube-ironic"}
-ssh -o "StrictHostKeyChecking=no" "core@$IP" sudo podman pull "${IRONIC_IMAGE}"
+echo -e "RHCOS_IMAGE_FILENAME_OPENSTACK=${RHCOS_IMAGE_FILENAME_OPENSTACK}\nIRONIC_IMAGE=${IRONIC_IMAGE}" \
+    | ssh -o "StrictHostKeyChecking=no" core@$IP sudo dd of=/etc/ironicservice
 
-ssh -o "StrictHostKeyChecking=no" core@$IP sudo podman run \
-    -d --net host --privileged --name ironic \
-    -v /home/core/dnsmasq.conf:/etc/dnsmasq.conf \
-    -v "/home/core/${RHCOS_IMAGE_FILENAME_OPENSTACK}:/var/www/html/images/${RHCOS_IMAGE_FILENAME_OPENSTACK}" \
-    "${IRONIC_IMAGE}"
+# Now that we have the Environment and the image, we can start the ironic service
+ssh -o "StrictHostKeyChecking=no" core@$IP sudo systemctl start ironic.service
 
 # Create a master_nodes.json file
 jq '.nodes[0:3] | {nodes: .}' "${WORKING_DIR}/ironic_nodes.json" | tee ocp/master_nodes.json
```

Comment:
ironic_nodes contains an IP on the default network, this need to be updated so the bootstrap node can contact ipmi
see
tripleo-quickstart-config/roles/libvirt/setup/overcloud/templates/ironic_nodes.json.j2:        "ipmi_address": "192.168.122.1",

Reply:
Thanks! I forgot about that

Reply:
Yeah this needs updating in https://github.com/metalkube/dev-scripts/blob/master/tripleo-quickstart-config/roles/libvirt/setup/overcloud/templates/ironic_nodes.json.j2#L20 I think or 06* wont' ever boot the nodes